src/loginPage/LoginPage.js

```diff
@@ -1,15 +1,72 @@
-import React from 'react';
+import React, {useState} from 'react';
+import {useNavigate} from 'react-router-dom';
+
+//TODO: replace to a separate file
+async function newSession(login, password) {
+    //TODO: replace to constants
+    const SERVER_PATH = 'http://localhost:5000';
+    const endPoint = SERVER_PATH + '/sessions/login';
+    let failedToFetch = false;
+
+    const response = await fetch(
+        endPoint,
+        { method: 'post',
+            body: JSON.stringify({ login, password }),
+            headers: { 'Content-type': 'application/json'}
+        })
+        .catch(() => failedToFetch = true);
+
+    if(failedToFetch){
+        return false;
+    }
+
+    if (response.ok){
+        const data = await response.json();
+        return { status: response.status, sessionID: data.sessionID, userRole: data.role };
+    }
+
+    return { status: response.status, massage: response.statusText };
+}
 
 export function LoginPage() {
+    const navigate = useNavigate();
+    const [login, setLogin] = useState('');
+    const [password, setPassword] = useState('');
+    const [authErrorText, setAuthErrorText] = useState('');
+
+    const handleLoginButtonClick = async () => {
+        const result = await newSession(login, password);
```

Comment:
Let me know if you do this because you prefer declarative style or because you are afraid of `.then` chains :)

Reply:
I prefer async/await syntax but in this case I forget that I can use `.then` chains

Reply:
Ok, just remember to be consequent in styling later